### PR TITLE
Test: Update Vagrant image to 107.

### LINKED ISSUE #5059: CI: VMs launched with test/Vagrantfile cannot use go build cache
Whenever I run `make` inside the VMs launched with `test/Vagrantfile`, e.g.:

1. `vagrant up runtime`
2. `vagrant ssh runtime`
3. `cd go/src/github.com/cilium/cilium`
4. `make`

I see output like the following:
```
go: disabling cache (/home/vagrant/.cache/go-build) due to initialization failure: open /home/vagrant/.cache/go-build/log.txt: permission denied
```

Fix this error and allow caching in these VMs for speeding up builds.

### PR DESCRIPTION
**Pull request by eloycoto**
Include the following PR:

https://github.com/cilium/packer-ci-build/pull/84
https://github.com/cilium/packer-ci-build/pull/82

Fix #5059

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5381)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: Vagrantfile [hunk 1]
```diff
@@ -128,7 +128,7 @@ Vagrant.configure(2) do |config|
         vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
 
         config.vm.box = "cilium/ubuntu-dev"
-        config.vm.box_version = "106"
```
Comment (aanm) [inline Vagrantfile:131]: There's no need to change this one as we didn't change anything relevant for developers.

**Review by aanm — CHANGES_REQUESTED** [review]

File: Vagrantfile (hunk 1 above)
Comment (eloycoto) [reply Vagrantfile:131]: There is a fix on `./cahe/.gobuild` so it's needed for developers, that make the build faster + don't show errors.

**Review by aanm — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Pull request merged**